spec/units/recruitment_banner_spec.rb

```diff
@@ -0,0 +1,138 @@
+      expect(described_class.all_banners.all? { |banner| validator_class.valid?(banner) }).to eq(true)
+    end
+
+    it "does not contain banners that are past their end date" do
+      expect(described_class.all_banners.all? { |banner| banner.end_date >= Time.zone.now }).to eq(true)
+    end
+
+    it "does not contain banners active at the same time and pointing to the same path" do
+      all_banners = described_class.all_banners
+      expect(all_banners.all? { |banner| validator_class.does_not_include_path_clashes?(banner, all_banners - [banner]) }).to eq(true)
+    end
+  end
+
+  context "test configurations" do
+    before do
+      original_path = Rails.root.join(__dir__, "../../config/govuk_web_banners/recruitment_banners.yml")
+      allow(YAML).to receive(:load_file).with(original_path).and_return(replacement_file)
+    end
+
+    context "with banners" do
+      let(:replacement_file) do
+        YAML.load_file(Rails.root.join(__dir__, "../../spec/fixtures/active_recruitment_banners.yml"))
+      end
+
+      describe ".for_path?" do
+        it "returns banner that includes the path" do
+          expect(described_class.for_path("/foreign-travel-advice")).to be_instance_of(GovukWebBanners::RecruitmentBanner)
+        end
+
+        it "returns nil for a path without a banner" do
+          expect(described_class.for_path("/foreign-climbing-advice")).to be_nil
+        end
+      end
+    end
+
+    context "with timed banners" do
+      let(:replacement_file) do
+        YAML.load_file(Rails.root.join(__dir__, "../../spec/fixtures/timed_recruitment_banners.yml"))
+      end
+
+      before { travel_to Time.local(2025, 1, 1) }
+      after { travel_back }
+
+      describe ".active_banners" do
+        it "returns an array with only banners that are active now" do
+          expect(described_class.active_banners.count).to eq(2)
+          expect(described_class.active_banners.first.name).to eq("Banner that has started and hasn't ended yet")
+          expect(described_class.active_banners.second.name).to eq("Banner that has started and has no end date")
+        end
+      end
+
+      describe ".all_banners" do
+        it "returns an array with all banners in the configuration " do
+          expect(described_class.all_banners.count).to eq(4)
+        end
+      end
+    end
+
+    context "with broken banners" do
+      let(:replacement_file) do
+        YAML.load_file(Rails.root.join(__dir__, "../../spec/fixtures/broken_recruitment_banners.yml"))
+      end
+
+      describe ".all_banners" do
+        it "confirms invalid banners exist" do
+          expect(described_class.all_banners.none? { |banner| validator_class.valid?(banner) }).to eq(true)
+        end
+      end
+    end
+
+    context "with expired banners" do
+      let(:replacement_file) do
+        YAML.load_file(Rails.root.join(__dir__, "../../spec/fixtures/expired_recruitment_banners.yml"))
+      end
+
+      describe ".all_banners" do
```

Comment:
Same here for the description of `all_banners`. This method just returns the banners.